src/charm.py

```diff
@@ -14,18 +14,19 @@
 class FastAPIDemoCharm(ops.CharmBase):
     """Charm the service."""
 
-    def __init__(self, framework):
+    def __init__(self, framework: ops.Framework) -> None:
         super().__init__(framework)
         self.pebble_service_name = "fastapi-service"
-        framework.observe(self.on.demo_server_pebble_ready, self._on_demo_server_pebble_ready)
+        framework.observe(self.on["demo-server"].pebble_ready, self._on_demo_server_pebble_ready)
 
-    def _on_demo_server_pebble_ready(self, event):
+    def _on_demo_server_pebble_ready(self, event: ops.PebbleReadyEvent) -> None:
         """Define and start a workload using the Pebble API.
 
         Change this example to suit your needs. You'll need to specify the right entrypoint and
         environment configuration for your specific workload.
 
-        Learn more about interacting with Pebble at at https://juju.is/docs/sdk/pebble.
+        Learn more about interacting with Pebble at https://juju.is/docs/sdk/pebble
+        Learn more about Pebble layers at https://github.com/canonical/pebble
```

Comment:
Wouldn't the [Layer doc](https://canonical-pebble.readthedocs-hosted.com/en/latest/reference/layers/) be a better place to link to for this?